tools/clang/test/HLSLFileCheck/hlsl/diagnostics/warnings/Wno-unsupported-statement-do-while-branch.hlsl

```diff
@@ -0,0 +1,17 @@
+// RUN: %dxc -T vs_6_0 -Wno-unsupported-statement %s | FileCheck %s
+
+// Make sure the specified warning gets turned off
+
+// This function has no output semantic on purpose in order to produce an error,
+// otherwise, the warnings will not be captured in the output for FileCheck.
+float main() {
+
+// attribute %0 can only be applied to 'if' and 'switch' statements
+// CHECK-NOT: statement
+  [branch]
+  do {} while(true);
```

Comment:
I think a test per attribute-statement combination is overkill when testing the -Wno-... options.

The warnings themselves should probably be tested for all attribute-statement combinations in `tools\clang\test\HLSL\attributes.hlsl` as part of the VerifierTest, such as:

https://github.com/microsoft/DirectXShaderCompiler/blob/f4c4c34662053deb41ba438ee2ea5092ec78f86f/tools/clang/test/HLSL/attributes.hlsl#L11

Here, I think it's sufficient to test turning off each warning group.

Reply:
Done